From d95281d7f248ba5aa31a8977909a3344ec5b132c Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Tue, 30 Jul 2019 08:15:59 +0200
Subject: [PATCH 1/1] add new channel after authentication

With libssh commit 8a885f0b ("channels: Add check if we are authenticated
before we create a channel") connection fails if channel is added before
successful authentication. So add the channel after authentication.

Fixes #154
---
 tmate-ssh-client.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/tmate-ssh-client.c b/tmate-ssh-client.c
index 96da14c8..2a9afee0 100644
--- a/tmate-ssh-client.c
+++ b/tmate-ssh-client.c
@@ -241,12 +241,6 @@ static void on_ssh_client_event(struct tmate_ssh_client *client)
 
 		ssh_set_callbacks(session, &client->ssh_callbacks);
 
-		client->channel = channel = ssh_channel_new(session);
-		if (!channel) {
-			tmate_fatal("cannot initialize");
-			return;
-		}
-
 		ssh_set_blocking(session, 0);
 		ssh_options_set(session, SSH_OPTIONS_HOST, client->server_ip);
 		ssh_options_set(session, SSH_OPTIONS_LOG_VERBOSITY, &verbosity);
@@ -370,6 +364,12 @@ static void on_ssh_client_event(struct tmate_ssh_client *client)
 		case SSH_AUTH_SUCCESS:
 			tmate_debug("Auth successful");
 			client->state = SSH_OPEN_CHANNEL;
+
+			client->channel = channel = ssh_channel_new(session);
+			if (!channel) {
+				tmate_fatal("cannot initialize");
+				return;
+			}
 			/* fall through */
 		}
 
